### DISCUSSION TITLE
Over the Air FSW update with F'

### DISCUSSION THREAD

**Discussion by rawinza555** [nasa/fprime#971] [Q&A] [answered]
Not sure if this is F' related. I watched a few videos on the mars helicopter mostly for inspiration and came across that you guys have a bug fix on mars right before the first flight. So im wondering what's you guys recommendation on FSW over the air update? Like uploading a new exeutable file to be run on the OBC. 

I get that I could upload the new executable file via fileupload function but I'm curious whether if it's possible to set up some sort of component on F' that would run a new executable on OBC and then if it has seg fault it would then run the previous "stable" version. 

Also not sure how the file transfer would work via rf transceiver. If anyone has a source recommendation on this please let me know.

**Comment by timcanham** [comment, marked as answer]
As you said, we just transferred the file using the `FileUplink` component, and then we ran a component specific to helicopter to install the binary. There is no universal way to do it. It is dependent on the how the platform boots, etc, but you can certainly write an F Prime component to do the update, just as we did.
